### PR TITLE
[New Rules] Veeam Credential Access DRs

### PR DESCRIPTION
**Pull request by w0rk3r**
## Summary

Adds 3 new rules to cover some TTPs described by [CrowdStrike](https://www.crowdstrike.com/blog/anatomy-of-alpha-spider-ransomware/) and [TheDFIRReport](https://thedfirreport.com/2021/12/13/diavol-ransomware/) as being abused by attackers to get access to Veeam credentials via MSSQL.

### CODE REVIEW

File: rules/windows/credential_access_veeam_backup_dll_imageload.toml [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+timestamp_override = "event.ingested"
+type = "eql"
+
+query = '''
+library where host.os.type == "windows" and event.action == "load" and
+  (dll.name : "Veeam.Backup.Common.dll" or dll.pe.original_file_name : "Veeam.Backup.Common.dll") and
+  (
+    process.code_signature.trusted == false or
+    process.code_signature.exists == false or
+    process.name : "powershell.exe"
```
Comment (Aegrah) [inline rules/windows/credential_access_veeam_backup_dll_imageload.toml:34]: Want to include scope to powershell_ise as well?

File: rules/windows/credential_access_veeam_commands.toml [hunk 2]
```diff
@@ -0,0 +1,70 @@
+[metadata]
+creation_date = "2024/03/14"
+integration = ["windows"]
```
Comment (Aegrah) [inline rules/windows/credential_access_veeam_commands.toml:3]: Should this one contain endpoint?

Comment (w0rk3r) [reply rules/windows/credential_access_veeam_commands.toml:3]: ++

File: rules/windows/credential_access_veeam_backup_dll_imageload.toml (hunk 1 above)
Comment (w0rk3r) [reply rules/windows/credential_access_veeam_backup_dll_imageload.toml]: Added

File: rules/windows/credential_access_veeam_commands.toml (hunk 2 above)
Comment (w0rk3r) [reply rules/windows/credential_access_veeam_commands.toml]: Added

File: rules/windows/credential_access_veeam_commands.toml [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+risk_score = 47
+rule_id = "b661f86d-1c23-4ce7-a59e-2edbdba28247"
+severity = "medium"
+tags = ["Domain: Endpoint", "OS: Windows", "Use Case: Threat Detection", "Tactic: Defense Evasion", "Tactic: Credential Access", "Data Source: Elastic Endgame", "Data Source: Elastic Defend"]
+timestamp_override = "event.ingested"
+type = "eql"
+
+query = '''
+process where host.os.type == "windows" and event.type == "start" and
+  process.args : "*[VeeamBackup].[dbo].[Credentials]*"
```
[the remark this answers is not in the document]
Comment (w0rk3r) [reply rules/windows/credential_access_veeam_commands.toml:32]: The process name is an attacker-controlled variable (Rename), so I don't see a lot of ways to do that
The one I used as a reference was sqlcmd, but there are a bunch of tools that can do that too

[the remark this answers is not in the document]
Comment (brokensound77) [reply rules/windows/credential_access_veeam_commands.toml:32]: can you share your test events for this one? Specifically curious on the full command line and args

[the remark this answers is not in the document]
Comment (w0rk3r) [reply rules/windows/credential_access_veeam_commands.toml:32]: Here is the SQL query:

> SELECT TOP (1000) [id],[user_name],[password],[usn],[description],[visible],[change_time_utc]FROM [VeeamBackup].[dbo].[Credentials];

Sqlcmd command-line:

```"C:\Program Files\Microsoft SQL Server\Client SDK\ODBC\130\Tools\Binn\sqlcmd.exe"  -S localhost,51341 -E -y0 -Q "SELECT TOP (1000) [id],[user_name],[password],[usn],[description],[visible],[change_time_utc]FROM [VeeamBackup].[dbo].[Credentials];"```

File: rules/windows/credential_access_veeam_backup_dll_imageload.toml [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+description = """
+Identifies potential credential decrypt operations by PowerShell or unsigned processes using the Veeam.Backup.Common.dll
+library. Attackers can use Veeam Credentials to target backups as part of destructive operations such as Ransomware
+attacks.
+"""
+from = "now-9m"
+index = ["logs-endpoint.events.*"]
+language = "eql"
+license = "Elastic License v2"
+name = "Potential Veeam Credential Decrypt"
```
Comment (Samirbous) [inline rules/windows/credential_access_veeam_backup_dll_imageload.toml:20]:
```suggestion
name = "Veeam Backup Library Loaded by Unusual Process"
```

File: rules/windows/credential_access_veeam_commands.toml (hunk 3 above)
[the remark this answers is not in the document]
Comment (Samirbous) [reply rules/windows/credential_access_veeam_commands.toml:32]: can you limit to `(process.name : "sqlcmd.exe" or process.pe.original_file_name == "SQLCMD.EXE")`

**Review by Samirbous — CHANGES_REQUESTED** [review]

File: rules/windows/credential_access_veeam_commands.toml (hunk 3 above)
[the remark this answers is not in the document]
Comment (w0rk3r) [reply rules/windows/credential_access_veeam_commands.toml:32]: Yeah, we can, but are there other clients that can perform the same query?

[the remark this answers is not in the document]
Comment (Samirbous) [reply rules/windows/credential_access_veeam_commands.toml:32]: for the sake of performance we can just use sqlcmd and add a note in the query comment to add more sql clients.

[the remark this answers is not in the document]
Comment (w0rk3r) [reply rules/windows/credential_access_veeam_commands.toml:36]: Updated the query with your condition and some PowerShell cmdlets

**Review by Samirbous — APPROVED** [review]
looking good thanks!

File: rules/windows/credential_access_veeam_backup_dll_imageload.toml [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+
+[rule]
+author = ["Elastic"]
+description = """
+Identifies potential credential decrypt operations by PowerShell or unsigned processes using the Veeam.Backup.Common.dll
+library. Attackers can use Veeam Credentials to target backups as part of destructive operations such as Ransomware
+attacks.
+"""
+from = "now-9m"
+index = ["logs-endpoint.events.*"]
```
Comment (terrancedejesus) [inline rules/windows/credential_access_veeam_backup_dll_imageload.toml:17]:
```suggestion
index = ["logs-endpoint.events.library*"]
```

File: rules/windows/credential_access_veeam_commands.toml [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+timestamp_override = "event.ingested"
+type = "eql"
+
+query = '''
+process where host.os.type == "windows" and event.type == "start" and
+  (
+    (process.name : "sqlcmd.exe" or process.pe.original_file_name : "sqlcmd.exe") or
+    process.args : ("Invoke-Sqlcmd", "Invoke-SqlExecute", "Invoke-DbaQuery", "Invoke-SqlQuery")
+  ) and
+  process.args : "*[VeeamBackup].[dbo].[Credentials]*"
```
Comment (terrancedejesus) [inline rules/windows/credential_access_veeam_commands.toml:36]:
```suggestion
  stringContains(process.args, "[VeeamBackup].[dbo].[Credentials]")
```

**Review by terrancedejesus — APPROVED** [review]
I left a couple suggestions that are easily reconcilable. Approving! Great work 🚀

File: rules/windows/credential_access_veeam_commands.toml (hunk 6 above)
Comment (w0rk3r) [reply rules/windows/credential_access_veeam_commands.toml:36]: Leaving this one out for now, until we do a ruleset-wide review after we can measure how effective this is

**Pull request merged**